R/unreachable_code_linter.R

```diff
@@ -37,24 +50,31 @@
 #'   linters = unreachable_code_linter()
 #' )
 #'
-#' code_lines <- "f <- if (foo) {\n 2 + 2\n}"
+#' code_lines <- "if (foo) {\n 2 + 2\n}"
 #' writeLines(code_lines)
 #' lint(
 #'   text = code_lines,
 #'   linters = unreachable_code_linter()
 #' )
 #'
-#' code_lines <- "f <- while (foo) {\n 2 + 2\n}"
+#' code_lines <- "while (foo) {\n 2 + 2\n}"
 #' writeLines(code_lines)
 #' lint(
 #'   text = code_lines,
 #'   linters = unreachable_code_linter()
 #' )
 #'
+#' code_lines <- "f <- function() {\n  return(1)\n  # end skip\n}"
+#' writeLines(code_lines)
+#' lint(
+#'   text = code_lines,
+#'   linters = unreachable_code_linter(allow_comment_regex = "# end skip")
+#' )
+#'
 #' @evalRd rd_tags("unreachable_code_linter")
 #' @seealso [linters] for a complete list of linters available in lintr.
 #' @export
-unreachable_code_linter <- function() {
+unreachable_code_linter <- function(allow_comment_regex = getOption("covr.exclude_end")) {
```

Comment:
explicit default if that option is unset?

Reply:
NULL is the default which I intended, upstream also writes the same, so I wouldn't write it out (do we have a linter proposed for supplying explicit arguments that match defaults?)

https://github.com/r-lib/covr/blob/54de614f5c17a42bfaef2c87dbf5723ca701326a/R/exclusions.R#L55-L57

Reply:
Actually, the defaults are set in `.onLoad`
https://github.com/r-lib/covr/blob/main/R/zzz.R

So the feature won't work unless we `loadNamespace("covr")`.
I suggest to default to this regex for nocov end.

Reply:
Hmm good point, thanks. We'll have to note about that in the docs.

Reply:
Sorry, I suggested to use the regex as default for `getOption(...)`. That way, we pick up custom exclusion patterns automatically while not depending on covr to be loaded for the default case.

Reply:
oh duh, that's obviously a way better option!